app/controllers/users/registrations_controller.rb

```diff
@@ -174,6 +175,35 @@ def create
     end
   end
 
+  def new_with_provider; end
+
+  def create_with_provider
+    @user = User.find_by_id(user_provider_params['user'])
+    # Create new team for the new user
+    @team = Team.new(team_provider_params)
+
+    if @team.valid? && @user.valid? && Rails.configuration.x.new_team_on_signup
```

Comment:
User.find_by_id returns a user from the database which is valid or nil so here will throw an error undefined method .valid? for nil class if the user is not found